addons/onboarding/tests/test_onboarding.py

```diff
@@ -215,3 +215,20 @@ def test_onboarding_step_without_onboarding(self):
         # But it can
         self.step_initially_w_o_onboarding.with_company(self.company_2).action_set_just_done()
         self.assert_onboarding_is_done(self.onboarding_3.with_company(self.company_2))
+
+    def test_remove_company_with_progress(self):
+        user = self.env['res.users'].create({
```

Comment:
You may import and use ``_mail_new_test_user`` tool to create users (include all necessary keys to avoid computation and speedup user creation, we don't need confirmation emails or stuff like that in tests :) )